### PR TITLE
Database

### PR DESCRIPTION
**Pull request by robDevs**
# Info
This PR finishes implementing the database and related functions. 

# Testing
Compile and run the code, use logcat to verify the database is refreshed and outputs fresh data. 

# Enable Logcat debugging
View->Tool windows->Logcat
In the Logcat window that change Verbose to Debug and enter "AppData_LOG" in the search bar (without quotes).

### CODE REVIEW

File: app/src/main/java/com/example/musclenerds/MainActivity.java [hunk 1]
```diff
@@ -85,4 +87,85 @@ public boolean onSupportNavigateUp() {
...
+        mDb.exerciseDAO().insert(newExercise);
+        newExercise = new Exercise("Push-Up",
+                "Get down on all fours, placing your hands slightly wider than your shoulders. Straighten your arms and legs. Lower your body until your chest nearly touches the floor. Pause, then push yourself back up. Repeat.",
+                "triceps",
+                4,
+                5,
+                "none",
+                "none for now");
+        mDb.exerciseDAO().insert(newExercise);
+    }
```
Comment (lastbestdev) [inline app/src/main/java/com/example/musclenerds/MainActivity.java:136]: I assume these functions are just hacks for now. That's fine.

But going forward, any ideas on how we could automate this better?

File: app/src/main/java/com/example/musclenerds/database/ExerciseDAO.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    List<Exercise> getAll();
+
+    @Insert
+    void insert(Exercise exercise);
+
+    @Update
+    void update(Exercise exercise);
+
+    @Delete
+    void delete(Exercise exercise);
```
Comment (lastbestdev) [inline app/src/main/java/com/example/musclenerds/database/ExerciseDAO.java:25]: ORMs are the 💣 . 😎

**Review by lastbestdev — APPROVED** [review]
We will definitely have to think about how we can automate populating the db going forward. 

The stuff you've got in this PR is huge and a bunch work. Thank you for taking point on this! Everything looking great. 👍

**Review by Shiviandemon — APPROVED** [review]

**Review by KyleV0318 — APPROVED** [review]

**Review by JosiahHamer2 — APPROVED** [review]

**Review by jseeley452 — APPROVED** [review]

**Review by Joshewah11 — COMMENTED** [review]
looks good to me

**Pull request merged**